### ISSUE TITLE
Relax some Checkstyle rules

### ISSUE THREAD

**Issue by ajs6f** [trellis-ldp/trellis#389]
I'm all in favor of Checkstyle rules to keep our code style coherent, but I think ours are a little too strict. We've discussed this for [specific examples](https://github.com/trellis-ldp/trellis/issues/311) before but I'd like to make some cases for other changes:

1. I'd like to allow imports using `*` (wildcard) period. There is nothing gained at all, in my opinion, by disallowing them. I don't know who traces type dependencies via reading `import` lines, but they shouldn't be. Use your IDE!
2. We require `final` almost anywhere it can be used. I know I probably contributed to this more than anyone, but now I've come to think this isn't adding any clarity, and modern JVMs no longer benefit at all from the optimization hints. Frankly, they are probably a lot better at figuring out when things do and don't change than are we-- they have runtime info.
3. We require braces for every statement. I would rather allow:
```
doSomething();
if (test == null) return Optional.empty();
doSomethingElse();
```
because I don't see how
```
doSomething();
if (test != null) {
    return Optional.empty();
}
doSomethingElse();
```
 is in any way better. If that seems too concise, we could allow braceless statements but require a new line and indent, a la:
```
doSomething();
if (test != null)
    return Optional.empty();
doSomethingElse();
```
which flows visually as does the braced version but saves a line.

**Comment by acoburn** [comment]
1. Permitting `*` imports is fine by me.
2. I see `final` more for my benefit as a developer rather than for the benefit of the compiler. That is, it's a check on me to ensure that I don't go and stupidly overwrite a value when I shouldn't be (a-là scala's `val` or JavaScript's `const` keywords). I'd be inclined to keep this as it is.
3. I don't have a strong opinion about the braces; that seems more aesthetic than anything else. I do like the clarity of including braces. Having spent way too much time writing Perl in the late 90s/early 00s, I've come to appreciate explicit and unambiguous control structures.

Mostly, I'd like stylistic consistency, and this configuration is pretty much just copied from other Apache projects.

**Comment by ajs6f** [comment]
I'm in favor of consistency, too, but for my money,the consistency comes from whitespace usage, not sticking `{` at the end of a line. I'll send a PR for the `*` imports, and what do you say to switching off the `{}` rule and seeing how bad it gets? :grin:

**Comment by acoburn** [comment]
Fine by me. The Star import config is [here](https://github.com/trellis-ldp/trellis/blob/master/buildtools/src/main/resources/checkstyle/checkstyle.xml#L69-L72). And I think just removing the `NeedBraces` config will do it for the control structures.

**Comment by acoburn** [comment]
Resolved with #391
